pkg/subctl/cmd/gather/resource.go

```diff
@@ -26,15 +26,16 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
-func ResourcesToYAMLFile(info *Info, ofType schema.GroupVersionResource, namespace string) {
+func ResourcesToYAMLFile(info *Info, ofType schema.GroupVersionResource, namespace string, listOptions metav1.ListOptions) {
 	err := func() error {
-		list, err := info.DynClient.Resource(ofType).Namespace(namespace).List(metav1.ListOptions{})
+		list, err := info.DynClient.Resource(ofType).Namespace(namespace).List(listOptions)
 		if err != nil {
 			return errors.WithMessagef(err, "error listing %q", ofType.Resource)
 		}
 
 		path := filepath.Join(info.DirName, info.ClusterName+"-"+ofType.Resource+".yaml")
-		file, err := os.Create(path)
+		// Create if new, else Append if file already exists
+		file, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_RDWR, 0666)
```

Comment:
Should we have different files for multiple instances of a resource? **For example** say
`gather.ConfigMapCoreDNS(info, "kube-system")` returns 5 configmaps. Do we want all 5 maps to be written to single file or separate file? I would say we write to separate file. Single file reading is not easy.

Reply:
Writing to separate files also makes it easier to compare dumps from multiple instances.

Reply:
Implemenation: https://github.com/submariner-io/submariner-operator/pull/1188

Reply:
Yeah, let's not append to already existing ones. I prefer the separate files, also by including the "namespace"

Otherwise multiple runs can become a problem, you don't know what's the right status. To correlate.

Reply:
Yeah, I would prefer multiple files and was going to bring it up for discussion.